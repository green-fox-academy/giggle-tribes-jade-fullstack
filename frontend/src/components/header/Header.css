body {
  margin: 0px;
}
/* Navbar container */
nav.header {
  display: flex;
  justify-content: space-between;
  overflow: hidden;
  background-color: #009587;
  font-family: Arial;
  font-weight: bold;
}

/* Title inside the navbar */
nav.header h1.header {
  margin: 0px;
  float: left;
  font-size: 30px;
  color: white;
  text-align: center;
  padding: 14px 16px;
  text-decoration: none;
}

/* Container of links inside the header */
nav.header div.header {
  display: flex;
  justify-content: flex-end;
}

/* Links inside the header */
nav.header div a {
  height: 60%;
  font-size: 16px;
  color: black;
  text-align: center;
  padding: 13% 16px 5px 16px;
  text-decoration: none;
}

/* Add a red background color to navbar links on hover */
nav.header div a.header:hover {
  color: white;
  background-color: #005e54;
}
