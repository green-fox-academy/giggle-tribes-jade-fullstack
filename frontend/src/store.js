import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';

import { userReducer } from './reducers/userReducer';
import { kingdomReducer } from './reducers/kingdomReducer';
import { buildingsReducer } from './reducers/buildingsReducer';
import { resourcesReducer } from './reducers/resourcesReducer';
import { troopsReducer } from './reducers/troopsReducer';
import { tokenReducer } from './reducers/tokenReducer';
import { errorReducer } from './reducers/errorReducer';

function root(state = {}, action) {
  return {
    user: userReducer(state.user, action),
    kingdom: kingdomReducer(state.kingdom, action),
    buildings: buildingsReducer(state.buildings, action),
    resources: resourcesReducer(state.resources, action),
    troops: troopsReducer(state.troops, action),
    token: tokenReducer(state.token, action),
    error: errorReducer(state.error, action),
  };
}

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
const store = createStore(
  root,
  /* preloadedState, */ composeEnhancers(applyMiddleware(thunk))
);

export { store };
