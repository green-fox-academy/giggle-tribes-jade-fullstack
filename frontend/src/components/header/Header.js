import React, { useEffect } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import './Header.css';
import HeaderTitle from './HeaderTitle';
import { setKingdomAction } from '../../actions/KingdomActions';

const loggedinHeaderItems = [
  { link: 'Settings', route: '/settings' },
  { link: 'Logout', route: '/login' },
];
const loggedoutHeaderItems = [
  { link: 'Login', route: '/login' },
  { link: 'Register', route: '/registration' },
];

const Header = ({ kingdom, token, kingdomName, set }) => {
  let headerItems = loggedoutHeaderItems;

  useEffect(() => {
    set(kingdom);
  }, [set, kingdom]);

  if (kingdomName.length > 0) {
    headerItems = loggedinHeaderItems;
  }
  return (
    <nav className="header">
      <HeaderTitle
        name={
          typeof kingdomName === 'string' && kingdomName.length > 0
            ? kingdomName
            : ''
        }
        route={
          typeof kingdomName === 'string' && kingdomName.length > 0
            ? '/kingdom/buildings'
            : ''
        }
      />
      <div className="header">
        {headerItems.map((item, index) => (
          <Link
            className="header"
            key={index.toString()}
            onClick={() => {
              if (item.link === 'Logout') localStorage.clear();
            }}
            to={item.route}
          >
            {item.link}
          </Link>
        ))}
      </div>
    </nav>
  );
};

Header.propTypes = {
  set: PropTypes.func.isRequired,
  kingdom: PropTypes.string.isRequired,
  token: PropTypes.string.isRequired,
  kingdomName: PropTypes.string,
};

const mapStateToProps = ({ token, kingdom, kingdomName }) => {
  return { token, kingdom, kingdomName };
};
const mapDispatchToProps = dispatch => {
  return {
    set: kingdomID => dispatch(setKingdomAction(kingdomID)),
  };
};
export default connect(mapStateToProps, mapDispatchToProps)(Header);
