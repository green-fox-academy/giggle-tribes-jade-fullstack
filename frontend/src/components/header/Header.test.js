import React from 'react';
import { Provider } from 'react-redux';
import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';
import configureStore from 'redux-mock-store';

import Header from './Header';
import { store } from '../../store';

let container;

beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it('renders Header without crashing without token', async () => {
  const mockStore = configureStore([]);
  const mockedStore = mockStore({
  });
  mockedStore.dispatch = jest.fn();
  Storage.prototype.getItem = jest.fn(key => {
  });
  await act(async () => {
    render(
    <Provider store={mockedStore}>
        <Header />
    </Provider>,
    container
    );
  });
});

it('renders Header without crashing with token but without kingdomName', async () => {
  const mockStore = configureStore([]);
  const mockedStore = mockStore({
  });
  mockedStore.dispatch = jest.fn();
  Storage.prototype.getItem = jest.fn(key => {
    return 'dummy_token';
  });
  await act(async () => {
    render(
    <Provider store={mockedStore}>
        <Header />
    </Provider>,
    container
    );
  });
});

it('renders Header without crashing with token and kingdomId', async () => {
  Storage.prototype.getItem = jest.fn(key => {
    return 'dummy_token';
  });
  const mockStore = configureStore([]);
  const mockedStore = mockStore({
    kingdom: 1
  });
  mockedStore.dispatch = jest.fn();
  await act(async () => {
    render(
    <Provider store={mockedStore}>
        <Header />
    </Provider>,
    container
    );
  });
});

it('matches snapshot without token', async () => {
  Storage.prototype.getItem = jest.fn(key => {
  });
  await act(async () => {
    render(
    <Provider store={store}>
        <Header />
    </Provider>,
    container
    );
  });
  expect(container.innerHTML).toMatchSnapshot();
});

it('matches snapshot with token without kingdomID', async () => {
  Storage.prototype.getItem = jest.fn(key => {
    return 'dummy_token';
  });
  const mockStore = configureStore([]);
  const mockedStore = mockStore({
  });
  mockedStore.dispatch = jest.fn();
  await act(async () => {
    render(
    <Provider store={mockedStore}>
        <Header />
    </Provider>,
    container
    );
  });
  expect(container.innerHTML).toMatchSnapshot();
});

it('matches snapshot with token and kingdomId', async () => {
  Storage.prototype.getItem = jest.fn(key => {
    return 'dummy_token';
  });
  const mockStore = configureStore([]);
  const mockedStore = mockStore({
    kingdom: 1,
  });
  mockedStore.dispatch = jest.fn();
  await act(async () => {
    render(
    <Provider store={mockedStore}>
        <Header />
    </Provider>,
    container
    );
  });
  expect(container.innerHTML).toMatchSnapshot();
});
