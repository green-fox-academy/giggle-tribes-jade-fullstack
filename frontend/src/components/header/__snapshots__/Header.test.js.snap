// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot with token and kingdomId 1`] = `"<nav class=\\"header\\"><a class=\\"header\\" href=\\"/kingdom/buildings\\"><h1 class=\\"header\\">Mocked Kingdom</h1></a><div class=\\"header\\"><a class=\\"header\\" href=\\"/settings\\">Settings</a><a class=\\"header\\" href=\\"/logout\\">Logout</a></div></nav>"`;

exports[`matches snapshot without token and kingdom 1`] = `"<nav class=\\"header\\"><h1 class=\\"header\\">Tribes of Vulpes</h1><div class=\\"header\\"><a class=\\"header\\" href=\\"/login\\">Login</a><a class=\\"header\\" href=\\"/registration\\">Register</a></div></nav>"`;
